### DISCUSSION TITLE
Multi-line descriptions for waypoints / POI / favorites

### DISCUSSION THREAD

**Discussion by s-n-ushakov** [osmandapp/OsmAnd#11798] [Ideas]
Might it be possible to implement consistent support for multi-line descriptions for waypoints / favorites / POI?

I am currently evaluating OsmAnd as Androzic replacement, and milti-line descriptions are important for me.

What I currently observe on OsmAnd version 3.9.10 is the following:
- it is possible to press the "Enter" virtual keyboard key while editing a favorites item description;
- a line break is inserted into the description and rendered in the text edit control appropriately;
- a bare line break is inserted in the text content of the related  /gpx/wpt/desc element in the "favourites.gpx" file;
- the issue: it is no more possible to edit the description, as every time the "Description" text edit control is tapped in the "Edit Favorite" dialog, the focus is immediately transferred to the "Name" control.

The device I use for evaluation is Samsung Galaxy S7 edge, Android 8.0.0.

**Comment by vshcherb** [comment]
Descriptions should already support multiline (at least editing and viewing). I'm not sure I can reproduce the issue. Is it possible to make a screen recording?

**Reply by s-n-ushakov** [reply]
Yes, viewing is supported, and editing is supported too. But the problem is that once the description becomes multi-line, it is only possible to edit it once :) All the subsequent attempts to edit the multi-line description result in edit position (the focus) being moved automatically to the "Name" edit control every time we tap the "Description" edit control.

Please see the screen recording with voice comments and a compressed sample "favourites.gpx" file attached:

https://user-images.githubusercontent.com/6242163/120438420-04e5bc00-c38a-11eb-9162-4aa63724dd20.mp4
[favourites.zip](https://github.com/osmandapp/OsmAnd/files/6582633/favourites.zip)

**Reply by vshcherb** [reply]
Thanks I see this is a bug.

**Reply by s-n-ushakov** [reply]
@vshcherb , thank you for quick review.
Should I file this bug as an issue formally now?
And could you kindly share your further vision on this topic? Is its complexity clear? Is it going to appear on some "todo" plan? Is it possible for me to get "subscribed" to the events related to this bug, especially to the fix and release events?

**Reply by vshcherb** [reply]
I've created a bug but we've already stuck with it and it's going to be closed. It looks like it's something related to your specific Android system or keyboard. Please try another keyboard.

**Reply by s-n-ushakov** [reply]
Thank you @vshcherb , and I can confirm that the latest OsmAnd build has no such problem with Samsung Note20. I can only guess what makes this difference.

Sorry for the confusion. And in case there are no more reports of issues like that, I guess the issue may be closed.
